packages/esm-system-admin-app/src/change-password-modal/change-password-modal.component.tsx

```diff
@@ -0,0 +1,203 @@
+      const lowercaseRegExp = /(?=.*?[a-z])/;
+      const digitsRegExp = /(?=.*?[0-9])/;
+      const minLengthRegExp = /.{8,}/;
+      const passwordLength = passwordInputValue.length;
+      const uppercasePassword = uppercaseRegExp.test(passwordInputValue);
+      const lowercasePassword = lowercaseRegExp.test(passwordInputValue);
+      const digitsPassword = digitsRegExp.test(passwordInputValue);
+      const minLengthPassword = minLengthRegExp.test(passwordInputValue);
+      let errMsg = '';
+      if (passwordLength === 0) {
+        errMsg = t('passwordIsEmpty', 'Password is empty');
+      } else if (!uppercasePassword) {
+        errMsg = t('atLeastOneUppercase', 'At least one Uppercase');
+      } else if (!lowercasePassword) {
+        errMsg = t('atLeastOneLowercase', 'At least one Lowercase');
+      } else if (!digitsPassword) {
+        errMsg = t('atLeastOneDigit', 'At least one digit');
+      } else if (!minLengthPassword) {
+        errMsg = t('minimum8Characters', 'Minimum 8 characters');
+      } else if (passwordInput.oldPassword.length > 0 && passwordInput.newPassword === passwordInput.oldPassword) {
+        errMsg = t('newPasswordMustNotBeTheSameAsOld', 'New password must not be the same as password');
+      } else {
+        errMsg = '';
+        setIsNewPasswordInvalid(false);
+      }
+      setNewPasswordErr(errMsg);
+    } else if (
+      passwordInputFieldName === 'confirmPassword' ||
+      (passwordInputFieldName === 'newPassword' && passwordInput.confirmPassword.length > 0)
+    ) {
+      if (passwordInput.confirmPassword !== passwordInput.newPassword) {
+        setConfirmPasswordError(t('confirmPasswordMustBeTheSameAsNew', 'Confirm password must be the same as new'));
+      } else {
+        setConfirmPasswordError('');
+        setIsConfirmPasswordInvalid(false);
+      }
+    } else {
+      if (passwordInput.newPassword.length > 0 && passwordInput.newPassword === passwordInput.oldPassword) {
+        setOldPasswordErr(t('oldPasswordMustNotBeTheSameAsNew', 'Old password must not be the same as new'));
+      } else {
+        setOldPasswordErr('');
+        setIsOldPasswordInvalid(false);
+      }
+    }
+  };
+
+  const handleSubmit = useCallback(
+    async (evt: React.FormEvent<HTMLFormElement>) => {
+      evt.preventDefault();
+      evt.stopPropagation();
+
+      try {
+        setIsSavingPassword(true);
+        await performPasswordChange(passwordInput.oldPassword, passwordInput.confirmPassword).then(() => {
+          close();
+          navigate({ to: `\${openmrsSpaBase}/logout` });
+          showToast({
+            title: t('userPassword', 'User password'),
+            description: t('userPasswordUpdated', 'User password updated successfully'),
+            kind: 'success',
+          });
+        });
+      } catch (error) {
+        setIsSavingPassword(false);
+        setErrorMessage(`${t('invalidPasswordCredentials', 'Invalid password provided')}: ${error.message}`);
+      }
```

Comment:
This construction is weird...

```ts
performPasswordChange(passwordInput.oldPassword, passwordInput.confirmPassword).then(() => {
  close();
   navigate({ to: `\${openmrsSpaBase}/logout` });
  showToast({
    title: t('userPassword', 'User password'),
    description: t('userPasswordUpdated', 'User password updated successfully'),
    kind: 'success',
  });
}).catch((err) => {
  setIsSavingPassword(false);
  setErrorMessage(`${t('invalidPasswordCredentials', 'Invalid password provided')}: ${error.message}`);
});
```

Reads better to me. (Basically _either_ use `await` and no `then()` or else just use it as a promise).

Reply:
The success message should be a snackbar, not a toast. Also, we should probably show a notification when it submits but fails at the backend.